File: sportify-standalone-application/src/main/java/com/spORtify/web/service/announcement/AnnouncementService.java
package com.spORtify.web.service.announcement;

import com.spORtify.data.entity.Announcement;
import com.spORtify.web.dto.AnnouncementDto;

import java.text.ParseException;
import java.util.List;


public interface AnnouncementService {
    List<Announcement> getAllAnnouncements();

    Announcement updateAnnouncement(String announcementToUpdateId, AnnouncementDto announcementDto) throws ParseException;

    void deleteAnnouncement(String id);

    void addAnnouncement(AnnouncementDto announcement) throws ParseException;

    void assignParticipantToAnnouncement(String announcementId, String userId);

    Announcement getAnnouncement(String announcementId);

}

File: sportify-frontend/src/service/announcementService.js
import axios from 'axios'
const baseUrl = 'http://localhost:8080/announcement'

const createAnnouncement = (newAnnouncement) => {
    const request = axios.post(`${baseUrl}/add`, newAnnouncement);
    return request.then(response => response.data);
}

const deleteAnnouncement = (announcementToDeleteId) => {
    const request = axios.delete(`${baseUrl}/delete/${announcementToDeleteId}`);
    return request.then(response => response.data);
}

const getAnnouncement = (id) => {
    const request = axios.get(`${baseUrl}/${id}`);
    return request.then(response => response.data);
}

const updateAnnouncement = (announcementToUpdateId, updatedAnnouncement) => {
    const request = axios.put(`${baseUrl}/update/${announcementToUpdateId}`, updatedAnnouncement);
    return request.then(response => response.data);
}


const assignParticipantToAnnouncement = (announcementId, userId) => {
    const request = axios.put(`${baseUrl}/assign/${announcementId}/${userId}`);
    return request.then(response => response.data);
}